Keas.Mvc/Handlers/ApiKeyMiddleware.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Keas.Core.Data;
+using Microsoft.AspNetCore.Http;
+
+namespace Keas.Mvc.Handlers
+{
+    public class ApiKeyMiddleware {
+        public const string HeaderKey = "X-Auth-Token";
+
+        private readonly RequestDelegate _next;
+
+        public ApiKeyMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+        public Task Invoke(HttpContext context, ApplicationDbContext dbContext)
+        {
+            // check for header
+            if (!context.Request.Headers.ContainsKey(HeaderKey))
+            {
+                return _next(context);
+            }
+            var headerValue = context.Request.Headers[HeaderKey].FirstOrDefault();
+
+            // lookup team that this API key has acess to
+            var teamAccess = dbContext.Teams.FirstOrDefault(t => t.ApiCode == new Guid(headerValue));
```

Comment:
@srkirkland I think we should also create a flag that says Ok to write. Either that, or create a new ApiCode for writing.